src/makeInfinityAdder.js

```diff
@@ -4,7 +4,22 @@
  * @return {function}
  */
 function makeInfinityAdder() {
-  // write code here
-}
+  let currentResult = 0;
+
+  function adder(number) {
+    if (number === undefined) {
+      const result = currentResult;
+
+      currentResult = 0;
+
+      return result;
```

Comment:
When `adder` is called without arguments, it should return the current sum and reset `currentResult`. This is correctly implemented, but adding a comment to explain this special case would improve code readability.